File: src/backend/Application/DTOs/MeterReadingUploadResultDto.cs
namespace Application.DTOs;

public class MeterReadingUploadResultDto
{
    public int TotalProcessed { get; set; }
    public int Successful { get; set; }
    public int Failed { get; set; }
    public List<MeterReadingDto> SuccessfulReadings { get; set; } = new();
    public List<MeterReadingUploadErrorDto> Errors { get; set; } = new();
}

File: src/frontend/src/types/MeterReadingUploadResultDto.ts
import type { MeterReadingDto } from './MeterReadingDto';
import type { MeterReadingUploadErrorDto } from './MeterReadingUploadErrorDto';

export interface MeterReadingUploadResultDto {
  totalProcessed: number;
  successful: number;
  failed: number;
  successfulReadings: MeterReadingDto[];
  errors: MeterReadingUploadErrorDto[];
}